pkg/executor/aggregate/agg_hash_partial_worker.go

```diff
@@ -116,43 +161,33 @@ func (w *HashAggPartialWorker) updatePartialResult(ctx sessionctx.Context, sc *s
 		return err
 	}
 
-	partialResults := w.getPartialResult(sc, w.groupKey, w.partialResultsMap)
+	finalWorkerIdxs, partialResults := w.getPartialResultsForAllFinalWorker(sc, w.groupKey, w.partialResultsMap, finalConcurrency)
```

Comment:
Use [][]aggfuncs.PartialResult is enough.   UpdatePartialResult don't care the finalWorkerIdx.

Reply:
> Use [][]aggfuncs.PartialResult is enough. UpdatePartialResult don't care the finalWorkerIdx.

finalWorkerIdx is necessary. When the i-th row in a chunk needs to be used for updating agg result, and we need to know which agg result is to be updated.

Reply:
> Use [][]aggfuncs.PartialResult is enough. UpdatePartialResult don't care the finalWorkerIdx.

done